RecoEgamma/EgammaTools/plugins/CandiateCombiner.cc

```diff
@@ -0,0 +1,120 @@
+#include "DataFormats/Common/interface/Ptr.h"
+#include "DataFormats/Math/interface/deltaR.h"
+
+class CandidateCombiner : public edm::stream::EDProducer<> {
+public:
+  explicit CandidateCombiner(const edm::ParameterSet&);
+  ~CandidateCombiner(){}
+  
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+  
+private:
+  void produce(edm::Event&, const edm::EventSetup&) override;
+
+private:  
+  edm::EDGetTokenT<edm::View<reco::Candidate> > coll1Token_;
+  edm::EDGetTokenT<edm::View<reco::Candidate> > coll2Token_;
+  const float maxDR2ToClean_;
+
+};
+
+
+namespace {
+  double pow2(double val){return val*val;}
+}
+
+CandidateCombiner::CandidateCombiner(const edm::ParameterSet& iConfig) :
+  coll1Token_(consumes<edm::View<reco::Candidate>>(iConfig.getParameter<edm::InputTag>("coll1"))),
+  coll2Token_(consumes<edm::View<reco::Candidate>>(iConfig.getParameter<edm::InputTag>("coll2"))),
+  maxDR2ToClean_(pow2(iConfig.getParameter<double>("maxDRToClean")))
+{
+  produces<std::vector<edm::Ptr<reco::Candidate> > >();
+}
+
+namespace {
+  template<typename T> edm::Handle<T> getHandle(const edm::Event& iEvent,
+						const edm::EDGetTokenT<T>& token)
+  {
+    edm::Handle<T> handle;
+    iEvent.getByToken(token,handle);
+    return handle;
+  }
+  
+  bool hasDRMatch(const reco::Candidate& cand,const std::vector<std::pair<float,float> >& etaPhisToMatch,const float maxDR2)
+  {
+    const float candEta = cand.eta();
+    const float candPhi = cand.phi();
+    for(const auto& etaPhi : etaPhisToMatch){
+      if(reco::deltaR2(candEta,candPhi,etaPhi.first,etaPhi.second)<=maxDR2){
+	return true;
+      }
+    }
+    return false;
+  }
+}
+
+// ------------ method called to produce the data  ------------
+void
+CandidateCombiner::produce(edm::Event& iEvent, const edm::EventSetup& iSetup)
+{
+  auto outColl = std::make_unique<std::vector<edm::Ptr<reco::Candidate> > >();
+  
+  auto coll1Handle = getHandle(iEvent,coll1Token_); 
+  auto coll2Handle = getHandle(iEvent,coll2Token_); 
+  
+  std::vector<std::pair<float,float> > coll1EtaPhis;
+  for(size_t objNr=0;objNr<coll1Handle->size();objNr++){
+    edm::Ptr<reco::Candidate> objPtr(coll1Handle,objNr);
+    coll1EtaPhis.push_back({objPtr->eta(),objPtr->phi()}); //just to speed up the DR match
+    outColl->push_back(objPtr);
+  }
+  for(size_t objNr=0;objNr<coll2Handle->size();objNr++){
+    edm::Ptr<reco::Candidate> objPtr(coll2Handle,objNr);
+    if(!hasDRMatch(*objPtr,coll1EtaPhis,maxDR2ToClean_)){
+      outColl->push_back(objPtr);
+    }
+  }
+    
+  iEvent.put(std::move(outColl));
+}
+
+// ------------ method fills 'descriptions' with the allowed parameters for the module  ------------
+void
+CandidateCombiner::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  
+  edm::ParameterSetDescription desc;
+  desc.add<edm::InputTag>("coll1", edm::InputTag("gedPhotons"));
+  desc.add<edm::InputTag>("coll2", edm::InputTag("gedGsfElectrons"));
```

Comment:
the description ```  +// usecase: getting a unified list of e/gammas + jets for jet core ```
doesn't seem to match these default values.

Reply:
default values updated